lib/bindConnectors/bindMessage.js

```diff
@@ -1,15 +1,111 @@
-var when             = require('when');
-var _ 				 = require('lodash');
+const when              = require('when');
+const _ 				  = require('lodash');
+const { performance } = require('perf_hooks');
+const { client, v2 }  = require('@datadog/datadog-api-client');
+
 
 const { camelCase }  = require('../utils/mout');
-var getMissingParams = require('./getMissingParams');
-var formatInputBody  = require('./formatInputBody');
+const getMissingParams = require('./getMissingParams');
+const formatInputBody  = require('./formatInputBody');
 const interceptAfterSuccess = require('./interceptAfterSuccess');
+const { initialiseLogger } = require('../logger');
+const getDatadogSecret = require('./getDatadogSecret');
 
 
-module.exports = function (message, threadneedle, connectorConfig) {
+const initDataDogApi = async () => {
+	const datadogToken = await getDatadogSecret();
+	if (datadogToken === undefined) {
+		return { datadogApiInstance: undefined, datadogToken: undefined };
+	}
+	try {
+		const configurationOpts = {
+			authMethods: {
+				apiKeyAuth: datadogToken,
+			},
+
+		};
+
+		const configuration = client.createConfiguration(configurationOpts);
+
+		if (process.env.DATADOG_HOST != null) {
+			client.setServerVariables(configuration, {
+				site: process.env.DATADOG_HOST
+			});
+		}
+
+
+		const datadogApiInstance = new v2.MetricsApi(configuration);
+		const logger = initialiseLogger(datadogToken);
+
+		return { datadogApiInstance, logger } ;
+	} catch (e) {
```

Comment:
[nit] no single char vars

and are we not logging out the error?

Reply:
fixed JB, thanks for spotting.